lib/grape/jsonapi/services/content_negotiator.rb

```diff
@@ -0,0 +1,50 @@
+module Grape
+  module Jsonapi
+    module Services
+      class ContentNegotiator
+        VALID_MEDIA_TYPE = 'application/vnd+json'.freeze
+
+        ERRORS = {
+          not_acceptable: {
+            status: 406,
+            message: 'Not Acceptable'
+          },
+          unsupported_media: {
+            status: 415,
+            message: 'Unsupported Media Type'
+          }
+        }.freeze
+
+        def initialize(api_version, headers)
+          @api_version    = api_version
+          @accept_version = headers['Accept-Version']
+          @content_type   = headers['Content-Type']
+          @accept_header  = headers['Accept']
+        end
+
+        def self.run(*args)
+          new(*args).run
+        end
+
+        def run
```

Comment:
@chandeeland I'm not super pleased with this returning `true`, `false`, or error. I think `true` or error would be better, I left a long commit message about it here: https://github.com/chandeeland/grape-jsonapi_entity/pull/15/commits/d90f91871332ff9d952d421e133cc15835613073.

We could also try moving the error handling entirely into the API, but I think it's better here. 

Also this might be full of holes so please poke it a few times. 

For reference: 
- JSON API specifications on content negotiation: http://jsonapi.org/format/#content-negotiation
- Grape documentation on using the Accept-Version header https://github.com/ruby-grape/grape#accept-version-header